subsys/net/ip/pmtu.c

```diff
@@ -0,0 +1,288 @@
+/** @file
+ * @brief IPv4/6 PMTU related functions
+ */
+
+/*
+ * Copyright (c) 2024 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(net_pmtu, CONFIG_NET_PMTU_LOG_LEVEL);
+
+#include <zephyr/kernel.h>
+#include <zephyr/net/net_event.h>
+#include <zephyr/net/net_mgmt.h>
+#include <zephyr/net/net_if.h>
+#include "pmtu.h"
+
+#if defined(CONFIG_NET_IPV4_PMTU)
+#define NET_IPV4_PMTU_ENTRIES CONFIG_NET_IPV4_PMTU_DESTINATION_CACHE_ENTRIES
+#else
+#define NET_IPV4_PMTU_ENTRIES 0
+#endif
+
+#if defined(CONFIG_NET_IPV6_PMTU)
+#define NET_IPV6_PMTU_ENTRIES CONFIG_NET_IPV6_PMTU_DESTINATION_CACHE_ENTRIES
+#else
+#define NET_IPV6_PMTU_ENTRIES 0
+#endif
+
+#define NET_PMTU_MAX_ENTRIES (NET_IPV4_PMTU_ENTRIES + NET_IPV6_PMTU_ENTRIES)
```

Comment:
Minor hidden detail, the number of cache entries each get a maximum option setting, however AFAICS the entries are shared between IPv4 and IPv6?

So if both are set to 3, you could have 6 IPv4 and 0 IPv6 entries? Or am I missing something?